src/components/Search/Panel/SearchPanel.js

```diff
@@ -41,36 +41,46 @@ class SearchPanel extends React.Component<P, {}> {
   constructor(props: P) {
     super(props);
 
-    this.segment = findParam('segment');
-    if (this.segment == null) {
-      this.segment = SEARCH_SEGMENT.BIBLIOGRAPHIC;
-    }
+    const { data: { search: { segment } } } = props;
+    const paramSegment = findParam('segment');
+
     this.state = {
       isBrowseRequested: false,
       searchForm: [EMPTY_STRING],
       filterEnable: true,
-      counter: [{}],
-      segment: this.segment,
+      segment: segment === undefined
+        ? paramSegment === null ? SEARCH_SEGMENT.BIBLIOGRAPHIC : paramSegment
+        : segment,
       btnSubmitEnabled: false
     };
 
     this.handleKeyDown = this.handleKeyDown.bind(this);
-    this.handleAddSearchForm = this.handleAddSearchForm.bind(this);
-    this.handleRemoveSearchForm = this.handleRemoveSearchForm.bind(this);
     this.handleOnChange = this.handleOnChange.bind(this);
-    this.handleResetAllButton = this.handleResetAllButton.bind(this);
   }
 
 
   componentDidMount() {
+    const { store: { getState } } = this.props;
+    const { btnSubmitEnabled } = this.state;
     const id = findParam('id') || findParam('savedId') || findParam('recordid');
+    if (!btnSubmitEnabled) {
+      const form = getState().form.searchForm;
+      if (typeof (form) !== 'undefined' && typeof (form.values) !== 'undefined' && typeof (form.values.searchTextArea) !== 'undefined') {
```

Comment:
Can just say `form?.values?.searchTextArea?.length` to reduce these 2 lines down to 1?

Reply:
I didn't know that possibility to simplify, thanks